File: webwork-web/web-core/src/types/Pageable.ts
/** 分页条件类型定义 */
export default class Pageable {
    // 当前页码
    curr = 1;
    // 每页条数
    size = 10;
    // 用户ID
    userId?: string;
    // 排序字段
    sortField?: string;
    // ASC DESC
    sortOrder?: string;
    // 其它条件
    condition?: any;
}

File: webwork-schema/src/main/java/com/github/wephotos/webwork/http/Pageable.java
package com.github.wephotos.webwork.http;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

/**
 * 用于分页的参数
 *
 * @author TQ
 */
@Getter
@Setter
@ToString
public class Pageable<T> {
    /**
     * 当前页码
     */
    private int curr = 1;
    /**
     * 起始下标
     */
    private int start = 0;
    /**
     * 每页条数
     */
    private int size = 15;

    /**
     * 用户ID
     */
    private String userId;
    /**
     * 查询条件
     */
    private T condition;
    /**
     * 排序字段
     */
    private String sortField;
    /**
     * ASC DESC
     */
    private String sortOrder;
}
